### ISSUE TITLE
What should default webhook payload be?

### ISSUE THREAD

**Issue by nicpottier** [nyaruka/floweditor#562]
Obviously when we migrate we create default bodies that are the same as our old ones, but the question is what should our new default be for posts?

Currently it is:

```
{
    "flow": @(json(run.flow)),
    "path": @(json(run.path)),
    "results": @(json(run.results)),
    "input": @(json(run.input)),
    "channel": @(json(run.input.channel))
    "contact": @(json(contact))
}
```

Which has the nice property of being simple looking.

However, it does really lock down our contract on the context. IE, we will be changing payloads if/when we change what is in say `run.flow` or `contact`. And because we don't know which fields are specifically being used, we can't rewrite these things dynamically.

Maybe we should choose something with less data but that is more explicit? Say like:

```
{
    "flow": @(json(object("uuid", run.flow.uuid, "name", run.flow.name))),
    "results": @(json(run.results)),
    "input": @(json(run.input.text)),
    "channel": @(json(object("uuid", run.input.channel.uuid, "name", run.input.channel.name)))
    "contact": @(json(object("uuid", contact.uuid, "name", contact.name)))
}
```

**Comment by nicpottier** [comment]
@nyaruka/nyaruka

**Comment by rowanseymour** [comment]
I'd definitely err on the side of less data and let users add what they need.

Just noticed we can use `object()` at the top-level too, so:

```
json(object(
    "flow", object("uuid", run.flow.uuid, "name", run.flow.name),
    "results", run.results,
    "input", run.input.text,
    "channel", object("uuid", run.input.channel.uuid, "name", run.input.channel.name),
    "contact", object("uuid", contact.uuid, "name", contact.name)
))
```

**Comment by ericnewcomer** [comment]
Anytime our context changes, webhook payloads can change since they can be arbitrary. For this reason we might even warn users against using embeds like `@(json(contact))`. In this case, our default/example payload should reflect that.  I definitely like results being the only top level dump since we don't really have much of a choice there. Basically if you do @(json(object)) then YMMV.

**Comment by nicpottier** [comment]
*important question* is it less data of fewer data?!?

**Comment by ericnewcomer** [comment]
Of course the scientists say fewer data. 
http://www.biomedicaleditor.com/word-usage-less.html

I've never liked data as a plural, nor as a countable noun. I think I'll just have to refer to it in bits and bytes so I can avoid it altogether.

**Comment by ericnewcomer** [comment]
So, to be clear, input is top level, so its `@input.channel.uuid`, etc. One issue is that input can be null, so we would need to wrap the whole thing in an if which is quite a bit uglier

```
"channel", if(input, object("uuid", input.channel.uuid, "name", input.channel.name), null)
```

**Comment by ericnewcomer** [comment]
Actually same goes for `input.text`

**Comment by rowanseymour** [comment]
I think you can do `default(object("uuid", input.channel.uuid, ...), null)` (debateable if that's better), but, yeah any of those could error...

**Comment by nicpottier** [comment]
Weren't we talking about having a `@channel` item? How do you split by channel at the start of a flow that's a broadcast?

**Comment by rowanseymour** [comment]
You mean a contact with no affinity? `@contact.channel` is the channel we _would_ send to even if we've never had an incoming from that contact. 

I don't recall us talking about having `@channel` but if we do then seems we should just remove `@contact.channel` to save confusion... (but don't forget `@channel` still could be null itself if a contact has no sendable URNs)

**Comment by nicpottier** [comment]
Oh ok, I think `@contact.channel` was what I was thinking about. That's not no affinity necessarily right, it's just the channel we WOULD send to if we needed to ya? I'm fine with that staying below `@contact` I think.

**Comment by rowanseymour** [comment]
Ya - basically looks for the first sendable URN/channel pair

**Comment by nicpottier** [comment]
Anyways, to circle back, how about this as the default:

```
json(object(
    "flow", object("uuid", run.flow.uuid, "name", run.flow.name),
    "contact", object("uuid", contact.uuid, "name", contact.name, "urn", contact.urn)
    "results", run.results
))
```

**Comment by nicpottier** [comment]
Well shoot, `@contact.urn` isn't a thing. What's the top priority URN? Just `@contact.urns[0]`?

**Comment by rowanseymour** [comment]
it should be...?

also `results` might cause less surprises than `run.results` for now

**Comment by nicpottier** [comment]
Oh it does, NM, need to fix completion in the editor. Phew.

**Comment by nicpottier** [comment]
Ok so:

```
@(json(object(
    "flow", object("uuid", run.flow.uuid, "name", run.flow.name),
    "contact", object("uuid", contact.uuid, "name", contact.name, "urn", contact.urn)
    "results", results
)))
```

**Comment by ericnewcomer** [comment]
Addressed in #563
